src/auth/route.js

```diff
@@ -0,0 +1,77 @@
+import 'dotenv/config';
+import express from 'express';
+import querystring from 'querystring';
+import request from 'request';
+
+import config from './data';
+
+import { 
+  generateRandomState, 
+  getAuthBaseURL,
+  getClientURL
+} from './util';
+
+const router = express.Router();
+const stateKey = 'github-auth-state';
+
+router.get('/login', (req, res) => {
+  console.log('/login');
+  const state = generateRandomState(16);
+  res.cookie(stateKey, state);
+  
+  const queryString = querystring.stringify({
+    client_id: config.github.clientId,
+    redirect_uri: config.github.redirectUrl,
+    state
+  });
+  
+  const url = getAuthBaseURL('/authorize?' + queryString);
```

Comment:
use es6 template string
```suggestion
  const url = getAuthBaseURL(`/authorize?${queryString}`);